org.omg.sysml.interactive.tests/src/org/omg/sysml/interactive/tests/ModelLevelEvaluationTest.java

```diff
@@ -343,6 +343,52 @@ public void testSelfIsMetaclassEvaluation() throws Exception {
 		assertFalse(evaluateBooleanValue(instance, 
 				checkMetaclassFeature(instance, "p"), 
 				"@SysML::AttributeUsage"));
+
+		assertTrue(evaluateBooleanValue(instance, 
+				checkMetaclassFeature(instance, "p"), 
+				"hastype SysML::PartUsage"));
+		assertFalse(evaluateBooleanValue(instance, 
+				checkMetaclassFeature(instance, "p"), 
+				"hastype SysML::ItemUsage"));
+	}
+	
+	@Test
+	public void testReflectiveAccess() throws Exception {
+		SysMLInteractive instance = getSysMLInteractiveInstance();
+		process(instance, "abstract part def P { attribute x; }");
+		
+		assertTrue(evaluateBooleanValue(instance, null, "P.metadata[1] istype SysML::PartDefinition"));
```

Comment:
Well, it is not an assumption, but a fact, since this is a test that is set up with a specific model. Since `P` doesn't have any metadata annotations, the only metadata will be the metaclass feature.